### ISSUE TITLE
Sik-Ho Tang | Brief Review -- Self-Supervised Learning of Visual Features Through Embedding Images into Text Topic Spaces.

### ISSUE THREAD

**Issue by NorbertZheng** [NorbertZheng/read-papers#129]
Sik-Ho Tang. [Brief Review — Self-Supervised Learning of Visual Features Through Embedding Images into Text Topic Spaces
](https://sh-tsang.medium.com/brief-review-self-supervised-learning-of-visual-features-through-embedding-images-into-text-c6aa9d930038).

**Comment by NorbertZheng** [comment]
## Overview
TextTopicNet, Train CNN Using Text Topics.
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/ec180276-1286-461c-ac25-5f2b91f5ff42)
Illustrated Wikipedia articles about specific entities.

Self-Supervised Learning of Visual Features Through Embedding Images into Text Topic Spaces, TextTopicNet, by UAB, and IIIT Hyderabad. 2017 CVPR, Over 80 Citations.
Self-Supervised Learning, Image Classification, Object Detection.

 - Visual features are learnt by mapping between topics and images.

**Comment by NorbertZheng** [comment]
## TextTopicNet
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/bb78c0a0-83a6-41a1-b701-360544960d7b)
TextTopicNet Framework.

### Latent Dirichlet Allocation (LDA)
 - LDA algorithm is a generative statistical model of a text corpus to extract the topics.
 - LDA can be represented as a **three level hierarchical Bayesian model**.

Each topic is characterized by a probability distribution over words.

**Comment by NorbertZheng** [comment]
### Convolutional Neural Network (CNN)
Two CNNs are tried.
 - One is the 8 layers CNN [CaffeNet](https://medium.com/coinmonks/paper-review-of-alexnet-caffenet-winner-in-ilsvrc-2012-image-classification-b93598314160), a modified version of [AlexNet](https://medium.com/coinmonks/paper-review-of-alexnet-caffenet-winner-in-ilsvrc-2012-image-classification-b93598314160).
 - The other architecture is a 6 layers CNN resulting from removing the 2 first convolutional layers from [CaffeNet](https://medium.com/coinmonks/paper-review-of-alexnet-caffenet-winner-in-ilsvrc-2012-image-classification-b93598314160).

For learning to predict the target topic probability distributions, a sigmoid cross-entropy loss is minimized on the image dataset.

**Comment by NorbertZheng** [comment]
### Pretraining Dataset
The ImageCLEF 2010Wikipedia collection consists of 237,434 Wikipedia images and the Wikipedia articles that contain these images.

After filtering, the training data is composed of 100,785 images and 35,582 unique articles.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/757ccc1a-7dc9-4f24-8cb1-d44a7883af5c)
Top-5 most relevant words for 3 of the discovered topics by LDA analysis (left), and top-5 most relevant images for the same topics (right). Overall word frequency is shown in blue, and estimated word frequency within the topic in red.

We can observe that the most representative images for each topic present some regularities and thus allow CNN to learn discriminative features.

**Comment by NorbertZheng** [comment]
## Results
### Image Classification
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/a5767439-0be7-4ce9-a7c5-f45cba8b397c)
One vs. Rest linear SVM validation %mAP on PASCAL VOC2007 by varying number of topics of LDA.

After pretraining, features are extracted from top layers of the CNN and one vs. rest linear SVMs are trained for image classification in PASCAL VOC2007 dataset.

 - It is found that the best validation performance is obtained for 40 topics.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/7cf4ef2a-c322-4e7b-b687-9001545d6776)
PASCAL VOC2007 per-class average precision (AP) scores for the classification task with pool5 features.

From the above table, the proposed TextTopicNet using pool5 features are substantially more discriminative than the rest for the most difficult classes, see e.g. “bottle”, “pottedplant” or “cow”.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/45a7e795-353e-409b-b10e-e8ceb982e6c6)
PASCAL VOC2007 %mAP image classification.

TextTopicNet (COCO): corresponds to a model trained with MS-COCO images and their ground-truth caption annotations as textual content.

TextTopicNet (Wiki) features, learned from a very noisy signal, perform surprisingly well compared with the ones of the TextTopicNet (COCO) model.

**Comment by NorbertZheng** [comment]
Noisy but large dataset!!!

**Comment by NorbertZheng** [comment]
### Multimodal Image Retrieval
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/713ea3c8-d631-4da4-b635-fd8b37b680aa)
MAP comparison on Wikipedia dataset [29] with supervised (bottom) and unsupervised (middle) methods.

For retrieval, images and documents are projected into the learned topic space and the KL-divergence distance of the query (image or text) is computed with all the entities in the database.

The proposed self-supervised TextTopicNet outperforms unsupervised approaches, and has competitive performance to supervised methods without using any labeled data.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/f41ad83f-bfba-4473-93fd-d9ecf976ac33)
Top 4 nearest neighbors for a given query image image (left-most). Each row makes use of features obtained from different layers of TextTopicNet (without fine tuning). From top to bottom: prob, fc7, fc6, pool5.

The results are semantically close, although not necessarily visually similar.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/e24715de-8c1b-40f1-8e45-9b6fb79d548b)
Top 10 nearest neighbors for a given text query (from left to right: “airplane”, “bird”, and “horse”) in the topic space of TextTopicNet.

By leveraging textual semantic information, TextTopicNet learns a polysemic representation of images.

(There are also other results such as object detection results, please feel free to read the paper directly.)

**Comment by NorbertZheng** [comment]
Zero-Shot!!!

**Comment by NorbertZheng** [comment]
Self-supervised learning is done by utilizing text passage and image in Wikipedia.

**Comment by NorbertZheng** [comment]
## Reference
 - [2017 CVPR] [TextTopicNet] [Self-Supervised Learning of Visual Features Through Embedding Images into Text Topic Spaces](https://arxiv.org/abs/1705.08631).
